src/fields/field_fe.hh

```diff
@@ -195,23 +195,55 @@ private:
         unsigned int range_end_;
     };
 
+	/**
+	 * Helper class holds data of invalid values of all regions.
+	 *
+	 * If region contains invalid element value (typically 'not a number') is_nan_ flag is set to true
+	 * and other information (element id an value) are stored. Check of invalid values is performed
+	 * during processing data of reader cache and possible exception is thrown only if FieldFE is defined
+	 * on appropriate region.
+	 *
+	 * Data of all regions are stored in vector of RegionValueErr instances.
+	 */
+    class RegionValueErr {
+    public:
+        /// Constructor
+        RegionValueErr() : is_nan_(false) {}
```

Comment:
Just use two constructors: default constructor for the valid and similar to `set` method for invalid.